src/ngx_http_lua_headers.c

```diff
@@ -146,9 +161,36 @@ ngx_http_lua_ngx_req_raw_header(lua_State *L)
     dd("size: %d", (int) size);
 
     if (hc->nbusy) {
+#if defined(nginx_version) && nginx_version >= 1011011
+        if (hc->nbusy > prealloc_nbusy) {
+            if (busy_bufs_ptrs != NULL)
+                ngx_pfree(ngx_cycle->pool, busy_bufs_ptrs);
+
+            busy_bufs_ptrs = ngx_palloc(ngx_cycle->pool,
```

Comment:
It is generally a bad idea to use the nginx memory pool for this since relatively small memory blocks *never* get freed in `ngx_pfree`. We should use `ngx_alloc` and `ngx_free` here directly.